app/lib/package/backend.dart

```diff
@@ -1079,43 +1079,60 @@ class PackageBackend {
     _logger.info('Invalidating cache for package ${newVersion.package}.');
     await purgePackageCache(newVersion.package);
 
-    try {
-      final emailFuture = emailBackend.trySendOutgoingEmail(emailEntityToStore);
-
-      final latestVersionChanged = prevLatestStableVersion != null &&
-          package!.latestVersion != prevLatestStableVersion;
-      final latestPrereleaseVersionChanged =
-          prevLatestPrereleaseVersion != null &&
-              package!.latestPrereleaseVersion != prevLatestPrereleaseVersion;
-      // Let's not block the upload response on these. In case of a timeout, the
-      // underlying operations still go ahead, but the `Future.wait` call below
-      // is not blocked on it.
-      await Future.wait([
-        emailFuture,
-        // Trigger analysis and dartdoc generation. Dependent packages can be left
-        // out here, because the dependency graph's background polling will pick up
-        // the new upload, and will trigger analysis for the dependent packages.
-        jobBackend.triggerAnalysis(newVersion.package, newVersion.version),
-        jobBackend.triggerDartdoc(newVersion.package, newVersion.version),
-        // Trigger a new doc generation for the previous latest stable version
-        // in order to update the dartdoc entry and the canonical-urls.
-        if (latestVersionChanged)
-          jobBackend.triggerDartdoc(newVersion.package, prevLatestStableVersion,
-              shouldProcess: true),
-        // Reset the priority of the previous pre-release version.
-        if (latestPrereleaseVersionChanged)
-          jobBackend.triggerDartdoc(
-              newVersion.package, prevLatestPrereleaseVersion,
-              shouldProcess: false),
-      ]).timeout(Duration(seconds: 10));
-    } catch (e, st) {
-      final v = newVersion.qualifiedVersionKey;
-      _logger.severe('Error post-processing package upload $v', e, st);
-    }
+    // Let's not block the upload response on these post-upload tasks.
+    // The operations should either be non-critical, or should be retried
+    // automatically.
+    Timer.run(() async {
+      try {
+        await _postUploadTasks(
+          package,
+          newVersion,
+          outgoingEmail,
+          prevLatestStableVersion: prevLatestStableVersion,
+          prevLatestPrereleaseVersion: prevLatestPrereleaseVersion,
+        );
+      } catch (e, st) {
+        final v = newVersion.qualifiedVersionKey;
+        _logger.severe('Error post-processing package upload $v', e, st);
```

Comment:
Where will this log stuff go when we do this?

Reply:
I think this may be correlated with the request, and could go at the end of that batch, but I am just guessing here. It could be just a separate log entry.